checkpoint/keeper_milestone.go

```diff
@@ -0,0 +1,208 @@
+package checkpoint
+
+import (
+	"errors"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cmn "github.com/maticnetwork/heimdall/common"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+)
+
+var (
+	MilestoneKey          = []byte{0x20} // Key to store milestone
+	CountKey              = []byte{0x30} //Key to store the count
+	MilestoneNoAckKey     = []byte{0x40} //Key to store the NoAckMilestone
+	MilestoneLastNoAckKey = []byte{0x50} //Key to store the Latest NoAckMilestone
+	LastMilestoneTimeout  = []byte{0x60} //Key to store the Last Milestone Timeout
+)
+
+// AddMilestone adds milestone into final blocks
+func (k *Keeper) AddMilestone(ctx sdk.Context, milestone hmTypes.Milestone) error {
+	milestoneNumber := k.GetMilestoneCount(ctx) + 1 //GetCount gives the number of previous milestone
+
+	key := GetMilestoneKey(milestoneNumber)
+	if err := k.addMilestone(ctx, key, milestone); err != nil {
+		return err
+	}
+
+	pruningNumber := milestoneNumber - 100
```

Comment:
Could we extract a constant?

Reply:
Yes, But it used only at one place. Should I move this constant value to config file?

Reply:
Let's do it. At the moment it's not obvious what 100 stands for.

Reply:
Done